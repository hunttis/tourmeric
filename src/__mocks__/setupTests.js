import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line

configure({ adapter: new Adapter() });
