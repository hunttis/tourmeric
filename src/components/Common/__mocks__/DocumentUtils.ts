
export default function scrollIntoview(_elementId: string) {
  console.log('scrolling');
}

export function setListener(_eventType: string, _callback: () => {}) {
  console.log('adding listener');
}

export function removeClassFromHtml(className: string) {
  console.log('removing ', className);
}

export function addClassToHtml(className: string) {
  console.log('adding ', className);
}