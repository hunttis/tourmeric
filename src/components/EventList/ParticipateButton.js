import React from 'react';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import { participate, checkParticipation, cancelParticipation } from '../../api/eventApi';

export const ParticipateButton = ({
  userId, profile, eventId, participations, waitList,
}) => {
  const alreadyParticipated = checkParticipation(userId, eventId, participations);
  if (alreadyParticipated) {
    return (
      <div>
        <button className="cancelbutton button is-rounded is-danger" onClick={() => cancelParticipation(eventId, userId)}>
          <p><Translate id="cancelparticipate" /></p>
          <span className="icon">
            <i className="fas fa-sign-out-alt" />
          </span>
        </button>
      </div>);
  }

  if (profile.isLoaded && !profile.isEmpty) {
    return (
      <button className="participatebutton button is-rounded is-primary" onClick={() => participate(eventId, userId, profile.firstName, profile.lastName)}>
        {!waitList &&
          <p><Translate id="participate" /></p>
        }
        {waitList &&
          <p><Translate id="participateforwaitlist" /></p>
        }

        <span className="icon">
          <i className="fas fa-sign-in-alt" />
        </span>
      </button>);
  }
  return (<div className="button is-black" disabled><Translate id="signintoparticipate" /></div>);
};

ParticipateButton.propTypes = {
  userId: PropTypes.string,
  profile: PropTypes.object,
  eventId: PropTypes.string,
  participations: PropTypes.object,
  waitList: PropTypes.bool,
};
