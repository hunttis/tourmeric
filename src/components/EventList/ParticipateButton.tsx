import React from 'react';
import { FormattedMessage } from "react-intl";
import _ from 'lodash';
import { History } from 'history';
import { Dispatch, AnyAction } from 'redux';
import { participantCount, participate, checkParticipation, cancelParticipation } from '../../api/eventApi';
import { FirebaseProfile, Participation } from '../../models/ReduxState';
import { TourmericEvent } from '../../models/Events';

interface Props {
  events: { [key: string]: TourmericEvent };
  userId: string;
  profile: FirebaseProfile;
  eventId: string;
  participations: { [key: string]: Participation };
  history: History;
  dispatch: Dispatch;
  setReturnLocation: (returnLocation: string) => AnyAction;
}

export const ParticipateButton = ({
  profile, events, eventId, participations, userId, history, setReturnLocation, dispatch,
}: Props) => {

  const alreadyParticipated = checkParticipation(userId, eventId, participations);
  const eventContent = _.get(events, eventId, {});
  const maxParticipants = _.get(eventContent, 'playerSlots', 0);
  const currentParticipants = participantCount(eventId, participations);
  const eventFull = Boolean(maxParticipants && maxParticipants <= currentParticipants);

  if (alreadyParticipated) {
    return (
      <div>
        <button className="cancelbutton button is-rounded is-danger" onClick={() => cancelParticipation(eventId, userId)}>
          <p><FormattedMessage id="cancelparticipate" /></p>
          <span className="icon">
            <i className="fas fa-sign-out-alt" />
          </span>
        </button>
      </div>
    );
  }

  if (profile.isLoaded && !profile.isEmpty) {
    return (
      <button className="participatebutton button is-rounded is-primary" onClick={() => participate(eventId, userId, profile.firstName, profile.lastName)}>
        {!eventFull &&
          <p><FormattedMessage id="participate" /></p>
        }
        {eventFull &&
          <p><FormattedMessage id="participateforwaitlist" /></p>
        }

        <span className="icon">
          <i className="fas fa-sign-in-alt" />
        </span>
      </button>);
  }
  return (
    <button
      className="button is-black"
      onClick={async () => {
        await dispatch(setReturnLocation(history.location.pathname));
        history.push('/login');
      }}
    >
      <FormattedMessage id="signintoparticipate" />
    </button>);
};
