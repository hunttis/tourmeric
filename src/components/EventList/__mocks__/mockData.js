export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
