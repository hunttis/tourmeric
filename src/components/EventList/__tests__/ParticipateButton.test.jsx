import React from 'react';
import { shallow } from 'enzyme';
import { ParticipateButton } from '../ParticipateButton';
import { mockEventId, mockParticipations, mockUserId, mockProfile } from '../__mocks__/mockData';
import '../../../__mocks__/setupTests';

const eventApi = require('../../../api/eventApi');

describe('ParticipateButton tests', () => {

  beforeEach(() => {
    eventApi.checkParticipation = jest.fn();
    eventApi.participate = jest.fn();
    eventApi.cancelParticipation = jest.fn();
  });

  it('Compares ParticipateButton to snapshot when the user HAS NOT participated', () => {
    eventApi.checkParticipation = jest.fn().mockReturnValue(true);
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(0);

    const modal = shallow(<ParticipateButton
      userId={mockUserId}
      eventId={mockEventId}
      profile={mockProfile}
      participations={mockParticipations}
    />);

    expect(modal).toMatchSnapshot();
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(1);
  });

  it('Compares ParticipateButton to snapshot when the user HAS participated', () => {
    eventApi.checkParticipation = jest.fn().mockReturnValue(false);
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(0);

    const modal = shallow(<ParticipateButton
      userId={mockUserId}
      eventId={mockEventId}
      profile={mockProfile}
      participations={mockParticipations}
    />);

    expect(modal).toMatchSnapshot();
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(1);
  });


  it('Tries to click participation button', () => {
    eventApi.checkParticipation = jest.fn().mockReturnValue(false);
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(0);

    const modal = shallow(<ParticipateButton
      userId={mockUserId}
      eventId={mockEventId}
      profile={mockProfile}
      participations={mockParticipations}
    />);

    expect(modal).toMatchSnapshot();
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(1);
    modal.find('.participatebutton').simulate('click');
    expect(eventApi.participate).toHaveBeenCalledTimes(1);
  });

  it('Tries to click cancel participation button', () => {
    eventApi.checkParticipation = jest.fn().mockReturnValue(true);
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(0);

    const modal = shallow(<ParticipateButton
      userId={mockUserId}
      eventId={mockEventId}
      profile={mockProfile}
      participations={mockParticipations}
    />);

    expect(modal).toMatchSnapshot();
    expect(eventApi.checkParticipation).toHaveBeenCalledTimes(1);
    modal.find('.cancelbutton').simulate('click');
    expect(eventApi.cancelParticipation).toHaveBeenCalledTimes(1);
  });


});
