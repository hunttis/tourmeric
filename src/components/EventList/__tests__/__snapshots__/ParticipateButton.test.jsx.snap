// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipateButton tests Compares ParticipateButton to snapshot when the user HAS NOT participated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParticipateButton
    eventId="mockEventId1"
    participations={
      Object {
        "mockEventId1": Object {
          "mockUserId": Object {
            "comment": "Jee",
            "date": "2018-06-23T09:40:26.992Z",
            "firstName": "Test",
            "lastName": "Person",
            "userId": "mockUserId",
          },
        },
      }
    }
    profile={
      Object {
        "avatarUrl": "https://foo.com/picture",
        "dciNumber": "123512351112",
        "displayName": "Test Person",
        "email": "fake@fake.com",
        "firstName": "Test",
        "isEmpty": false,
        "isLoaded": true,
        "lastName": "Person",
        "providerData": Array [
          Object {
            "displayName": "Test Person",
            "email": "fake@fake.com",
            "photoURL": "https://foo.com/picture",
            "providerId": "facebook.com",
            "uid": "1234567890",
          },
        ],
        "role": "admin",
      }
    }
    userId="mockUserId"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="cancelbutton button is-rounded is-danger"
        onClick={[Function]}
      >
        <p>
          <Translate
            id="cancelparticipate"
          />
        </p>
        <span
          className="icon"
        >
          <i
            className="fas fa-sign-out-alt"
          />
        </span>
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <Translate
              id="cancelparticipate"
            />
          </p>,
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-out-alt"
            />
          </span>,
        ],
        "className": "cancelbutton button is-rounded is-danger",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Translate
              id="cancelparticipate"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "cancelparticipate",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-sign-out-alt"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sign-out-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="cancelbutton button is-rounded is-danger"
          onClick={[Function]}
        >
          <p>
            <Translate
              id="cancelparticipate"
            />
          </p>
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-out-alt"
            />
          </span>
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <Translate
                id="cancelparticipate"
              />
            </p>,
            <span
              className="icon"
            >
              <i
                className="fas fa-sign-out-alt"
              />
            </span>,
          ],
          "className": "cancelbutton button is-rounded is-danger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Translate
                id="cancelparticipate"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "cancelparticipate",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-sign-out-alt"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-sign-out-alt",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "span",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ParticipateButton tests Compares ParticipateButton to snapshot when the user HAS participated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParticipateButton
    eventId="mockEventId1"
    participations={
      Object {
        "mockEventId1": Object {
          "mockUserId": Object {
            "comment": "Jee",
            "date": "2018-06-23T09:40:26.992Z",
            "firstName": "Test",
            "lastName": "Person",
            "userId": "mockUserId",
          },
        },
      }
    }
    profile={
      Object {
        "avatarUrl": "https://foo.com/picture",
        "dciNumber": "123512351112",
        "displayName": "Test Person",
        "email": "fake@fake.com",
        "firstName": "Test",
        "isEmpty": false,
        "isLoaded": true,
        "lastName": "Person",
        "providerData": Array [
          Object {
            "displayName": "Test Person",
            "email": "fake@fake.com",
            "photoURL": "https://foo.com/picture",
            "providerId": "facebook.com",
            "uid": "1234567890",
          },
        ],
        "role": "admin",
      }
    }
    userId="mockUserId"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <Translate
            id="participate"
          />
        </p>,
        <span
          className="icon"
        >
          <i
            className="fas fa-sign-in-alt"
          />
        </span>,
      ],
      "className": "participatebutton button is-rounded is-primary",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Translate
            id="participate"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "participate",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-sign-in-alt"
          />,
          "className": "icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-sign-in-alt",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <Translate
              id="participate"
            />
          </p>,
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-in-alt"
            />
          </span>,
        ],
        "className": "participatebutton button is-rounded is-primary",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Translate
              id="participate"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "participate",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-sign-in-alt"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sign-in-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ParticipateButton tests Tries to click cancel participation button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParticipateButton
    eventId="mockEventId1"
    participations={
      Object {
        "mockEventId1": Object {
          "mockUserId": Object {
            "comment": "Jee",
            "date": "2018-06-23T09:40:26.992Z",
            "firstName": "Test",
            "lastName": "Person",
            "userId": "mockUserId",
          },
        },
      }
    }
    profile={
      Object {
        "avatarUrl": "https://foo.com/picture",
        "dciNumber": "123512351112",
        "displayName": "Test Person",
        "email": "fake@fake.com",
        "firstName": "Test",
        "isEmpty": false,
        "isLoaded": true,
        "lastName": "Person",
        "providerData": Array [
          Object {
            "displayName": "Test Person",
            "email": "fake@fake.com",
            "photoURL": "https://foo.com/picture",
            "providerId": "facebook.com",
            "uid": "1234567890",
          },
        ],
        "role": "admin",
      }
    }
    userId="mockUserId"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="cancelbutton button is-rounded is-danger"
        onClick={[Function]}
      >
        <p>
          <Translate
            id="cancelparticipate"
          />
        </p>
        <span
          className="icon"
        >
          <i
            className="fas fa-sign-out-alt"
          />
        </span>
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <Translate
              id="cancelparticipate"
            />
          </p>,
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-out-alt"
            />
          </span>,
        ],
        "className": "cancelbutton button is-rounded is-danger",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Translate
              id="cancelparticipate"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "cancelparticipate",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-sign-out-alt"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sign-out-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="cancelbutton button is-rounded is-danger"
          onClick={[Function]}
        >
          <p>
            <Translate
              id="cancelparticipate"
            />
          </p>
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-out-alt"
            />
          </span>
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <Translate
                id="cancelparticipate"
              />
            </p>,
            <span
              className="icon"
            >
              <i
                className="fas fa-sign-out-alt"
              />
            </span>,
          ],
          "className": "cancelbutton button is-rounded is-danger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Translate
                id="cancelparticipate"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "cancelparticipate",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-sign-out-alt"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-sign-out-alt",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "span",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ParticipateButton tests Tries to click participation button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParticipateButton
    eventId="mockEventId1"
    participations={
      Object {
        "mockEventId1": Object {
          "mockUserId": Object {
            "comment": "Jee",
            "date": "2018-06-23T09:40:26.992Z",
            "firstName": "Test",
            "lastName": "Person",
            "userId": "mockUserId",
          },
        },
      }
    }
    profile={
      Object {
        "avatarUrl": "https://foo.com/picture",
        "dciNumber": "123512351112",
        "displayName": "Test Person",
        "email": "fake@fake.com",
        "firstName": "Test",
        "isEmpty": false,
        "isLoaded": true,
        "lastName": "Person",
        "providerData": Array [
          Object {
            "displayName": "Test Person",
            "email": "fake@fake.com",
            "photoURL": "https://foo.com/picture",
            "providerId": "facebook.com",
            "uid": "1234567890",
          },
        ],
        "role": "admin",
      }
    }
    userId="mockUserId"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <Translate
            id="participate"
          />
        </p>,
        <span
          className="icon"
        >
          <i
            className="fas fa-sign-in-alt"
          />
        </span>,
      ],
      "className": "participatebutton button is-rounded is-primary",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Translate
            id="participate"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "participate",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-sign-in-alt"
          />,
          "className": "icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-sign-in-alt",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <Translate
              id="participate"
            />
          </p>,
          <span
            className="icon"
          >
            <i
              className="fas fa-sign-in-alt"
            />
          </span>,
        ],
        "className": "participatebutton button is-rounded is-primary",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Translate
              id="participate"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "participate",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-sign-in-alt"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sign-in-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
